index.js

```diff
@@ -5,26 +5,17 @@ export function includeKeys(object, predicate) {
 		const set = new Set(predicate);
 		for (const key of Object.keys(object)) {
 			if (set.has(key)) {
-				const value = object[key];
-				Object.defineProperty(result, key, {
-					value,
-					writable: true,
-					enumerable: true,
-					configurable: true,
-				});
+				const descriptor = Object.getOwnPropertyDescriptor(object, key);
+				Object.defineProperty(result, key, descriptor);
 			}
 		}
 	} else {
 		// `for ... of Object.keys()` is faster than `for ... of Object.entries()`.
 		for (const key of Object.keys(object)) {
 			const value = object[key];
```

Comment:
Note: `value` and `descriptor` are retrieved separately, since `descriptor.value !== value` when `descriptor.get` is present.